README.md

```diff
@@ -1,133 +1,616 @@
+# Yarn is used in this example, but equivalent commands using other package managers can be used, e.g. npm and pnpm
+yarn
+yarn build
+```
+
+> Note that running the NPM scripts to verify the generated TypeScript client is NOT part of the code generation process, and needs to be explicitly executed after the client is generated.
+
+### Using Smithy TypeScript with Gradle
+
+Using the Smithy CLI, a new Gradle project can be created using the `quickstart-gradle` template. In this example, the project will be called `smithy-typescript-example-client-gradle`.
+
+```shell
+smithy init -t quickstart-gradle -o smithy-typescript-example-client-gradle
+cd smithy-typescript-example-client-gradle/
+```
+
+This will create a project with the following directory structure:
+
+```text
+smithy-typescript-example-client-gradle/
+├── README.md
+├── build.gradle.kts
+├── gradle
+│   └── wrapper
+│       ├── gradle-wrapper.jar
+│       └── gradle-wrapper.properties
+├── gradle.properties
+├── gradlew
+├── gradlew.bat
+├── models
+│   └── weather.smithy
+├── settings.gradle.kts
+└── smithy-build.json
+```
+
+To add a minimal `typescript-client-codegen` plugin, add the following to `smithy-build.json`: 
+
+```json
+// smithy-build.json
+{
+    "version": "1.0",
+    "sources": ["models"],
+    "plugins": {
+        // Add the Smithy TypeScript client plugin
+        "typescript-client-codegen": {
+            // Minimal configuration: add package name and version
+            "package": {
+                "package": "@smithy/typescript-example-client",
+                "packageVersion": "0.0.1"
+            }
         }
     }
+}
+```
 
-    plugins {
-        id("software.amazon.smithy").version("0.6.0")
-    }
+> Note: Maven dependencies cannot be configured in `smithy-build.json` for Gradle projects.
 
-    repositories {
-        mavenLocal()
-        mavenCentral()
-    }
+Then, add the `smithy-typescript-codegen` dependency in `build.gradle.kts`:
 
-    dependencies {
-        implementation("software.amazon.smithy:smithy-model:[1.33.0,1.34.0[")
-        implementation("software.amazon.smithy.typescript:smithy-typescript-codegen:0.18.0")
-    }
-   ```
-
-3. Create a `smithy-build.json` file with the following contents,
-   substituting "smithy.example#ExampleService" with the name of the service
-   to generate and "smithyexample" with the name of the TypeScript package to
-   create.:
-
-   ```json
-    {
-        "version": "1.0",
-        "plugins": {
-            "typescript-codegen": {
-                "service": "smithy.example#ExampleService",
-                "targetNamespace": "SmithyExample",
-                "package": "smithyexample",
-                "packageVersion": "0.0.1",
-                "packageJson": {
-                    "license": "Apache-2.0"
-                }
-            }
-        }
-    }
+```kotlin
+plugins {
+    id("java-library")
+    id("software.amazon.smithy.gradle.smithy-jar").version("0.8.0")
```

Comment:
I think we just need the `smithy-base` plugin here b/c we dont actually care about the generated jar.

Reply:
This was from the `quickstart-gradle` template, so I don't think it makes sense to go into that much detail for a quick start.

Reply:
We should make sure when we add the new example templates for this repo they use the correct plugin. This one is not correct for this use case.